packages/mermaid/src/rendering-util/splitText.ts

```diff
@@ -0,0 +1,128 @@
+import type { CheckFitFunction, MarkdownLine, MarkdownWord, MarkdownWordType } from './types.js';
+
+/**
+ * Splits a string into graphemes if available, otherwise characters.
+ */
+export function splitTextToChars(text: string): string[] {
+  if (Intl.Segmenter) {
+    return [...new Intl.Segmenter().segment(text)].map((s) => s.segment);
+  }
+  return [...text];
+}
+
+/**
+ * Splits a string into words.
```

Comment:
It might be worth adding a comment here that this function may return different results depending on the environment/locale.

I think `Intl.Segmenter(undefined)` will use the default locale, which might be `en-US` on one browser, and something else on another browser.

From [**Why should we pass a locale and options bag for grapheme boundaries? Isn't there just one way to do it?**](https://github.com/tc39/proposal-intl-segmenter#why-should-we-pass-a-locale-and-options-bag-for-grapheme-boundaries-isnt-there-just-one-way-to-do-it), it sounds like `Intl.Segmenter` might start returning different results for some Indic languages like Bengali (e.g. `bn-IN` locale).